pkgs/development/python-modules/txaio/default.nix

```diff
@@ -1,23 +1,27 @@
-{ stdenv, buildPythonPackage, fetchurl,
-  pytest, mock, six, twisted
-}:
+{ stdenv, buildPythonPackage, fetchPypi, pytest, mock, six, twisted }:
+
 buildPythonPackage rec {
   name = "${pname}-${version}";
   pname = "txaio";
   version = "2.7.1";
 
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1lmllmjjsqzl3w4faq2qhlgkaqn1yn1m7d99k822ib7qgz18bsly";
+  };
+
   buildInputs = [ pytest mock ];
+
   propagatedBuildInputs = [ six twisted ];
 
+  patchPhase = ''
```

Comment:
if you override the patchPhase, you remove the test runner. Use `postPatch` here instead because you patch code.